### PR TITLE
Resource estimation from llops

### PR DESCRIPTION
**Pull request by gwwatkin**
Changes to make the resource estimator work from Logical Lattice Ops (similar to topological assembly). 

The resource estimation from slices was incorrect. This is a more accurate use of Qentiana. Been using it in the benchmarks and getting reasonable results. Also went over it with @alexandrupaler.

### CODE REVIEW

File: src/lsqecc/resource_estimation/llops_resource_estimator.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    qentiana = Qentiana(ex1.props)
+
+    try:
+        results = qentiana.compute_physical_resources()
+        return ResourcesEstimatedFromLLOPS(
+            code_distance=results["distance"], time_secs=results["time"]
+        )
+    except TypeError:
+        return None
```
Comment (gwwatkin) [inline src/lsqecc/resource_estimation/llops_resource_estimator.py:70]: @alexandrupaler this is how we are using Qentiana

File: src/lsqecc/resource_estimation/slice_resource_estimator.py [hunk 2]
```diff
@@ -44,34 +44,32 @@ class EstimatedResources:
...
-
 
 def estimate_resources(
     computation: lscc.LatticeSurgeryComputation,
     # Device features
     code_distance: int = 7,  # NOTE: this value matches the 7 in qentiana
     error_decoding_time_ns: float = (10**3),
     decoding_time_by_code_distance_multiplier: float = 1,
-    physical_error_rate: float = 0.001,  # As expressed in quentiana, TODO find units
+    physical_error_rate: float = 0.0001,  # As expressed in quentiana, TODO find units
```
Comment (gwwatkin) [inline src/lsqecc/resource_estimation/slice_resource_estimator.py]:
```suggestion
    physical_error_rate: float = 0.001,  # As expressed in quentiana, TODO find units
```

File: src/lsqecc/resource_estimation/llops_resource_estimator.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+class ResourcesEstimatedFromLLOPS:
+    code_distance: int = 7  # update from
+    time_secs: float = 0
+
+
+#
+# vvvv THIS IS WHAT WE WANT TO COMPLETE TODAY vvvv
+#
+# Left some TODOs
+#
```
Comment (gwwatkin) [inline src/lsqecc/resource_estimation/llops_resource_estimator.py]:
```suggestion
```

File: src/lsqecc/resource_estimation/llops_resource_estimator.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+
+@dataclass
+class ResourceEstimationConfig:
+    # From qentiana:
+    # target_error_per_T_gate = 1 / (self.parameters["safety_factor"] * local_t_count)
+    # TODO why do we prefer to specify it in this way, instead of a global target error rate?
+    safety_factor: float = 99
+    physical_error_rate: float = 0.001
+    prefer_depth_over_t_count = True  # TODO decide
```
Comment (gwwatkin) [inline src/lsqecc/resource_estimation/llops_resource_estimator.py]:
```suggestion
```

**Review by alexandrupaler — APPROVED** [review]

**Comment by gwwatkin** [comment]
pre-commit.ci autofix

**Pull request merged**